projects/components/src/table/data/table-cdk-data-source.ts

```diff
@@ -214,6 +214,7 @@ export class TableCdkDataSource implements DataSource<TableRow> {
       map(response => response.data),
       map(rows => this.paginateRows(rows, pageEvent)),
       map(TableCdkRowUtil.buildInitialRowStates),
+      map(TableCdkRowUtil.unselectAllRows),
```

Comment:
Why is this needed? Wouldn't this clear initial selections?

Reply:
I added that because rows were getting randomly selected while navigating through multiple table pages through paginator. 

But, it seems to be fixed now after @jake-bassett's latest fix for multi select on the table